Lets Do This/Controllers/User/LDTUserViewController.m

```diff
@@ -0,0 +1,240 @@
+        UIAlertController *reportbackPhotoAlertController = [UIAlertController alertControllerWithTitle:@"Pics or it didn't happen!" message:nil                                                              preferredStyle:UIAlertControllerStyleActionSheet];
+        UIAlertAction *cameraAlertAction;
+        if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+            cameraAlertAction = [UIAlertAction actionWithTitle:@"Take Photo" style:UIAlertActionStyleDefault handler:^(UIAlertAction * action){
+                [[GAI sharedInstance] trackEventWithCategory:@"behavior" action:@"choose reportback photo" label:@"camera" value:nil];
+                self.imagePickerController.sourceType = UIImagePickerControllerSourceTypeCamera;
+                [tabBarController presentViewController:self.imagePickerController animated:YES completion:NULL];
+            }];
+        }
+        else {
+            cameraAlertAction = [UIAlertAction actionWithTitle:@"(Camera Unavailable)" style:UIAlertActionStyleDefault handler:^(UIAlertAction * action){
+                // Nada
+            }];
+        }
+        UIAlertAction *photoLibraryAlertAction = [UIAlertAction actionWithTitle:@"Choose From Library" style:UIAlertActionStyleDefault handler:^(UIAlertAction * action){
+            [[GAI sharedInstance] trackEventWithCategory:@"behavior" action:@"choose reportback photo" label:@"gallery" value:nil];
+            self.imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+            [tabBarController presentViewController:self.imagePickerController animated:YES completion:NULL];
+        }];
+        UIAlertAction *cancelAlertAction = [UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:^(UIAlertAction * action) {
+            [reportbackPhotoAlertController dismissViewControllerAnimated:YES completion:nil];
+        }];
+        [reportbackPhotoAlertController addAction:cameraAlertAction];
+        [reportbackPhotoAlertController addAction:photoLibraryAlertAction];
+        [reportbackPhotoAlertController addAction:cancelAlertAction];
+        [tabBarController presentViewController:reportbackPhotoAlertController animated:YES completion:nil];
+    });
+}
+
+- (IBAction)settingsTapped:(id)sender {
+    LDTSettingsViewController *destVC = [[LDTSettingsViewController alloc] initWithNibName:@"LDTSettingsView" bundle:nil];
+    UINavigationController *destNavVC = [[UINavigationController alloc] initWithRootViewController:destVC];
+    [destNavVC styleNavigationBar:LDTNavigationBarStyleClear];
+    LDTTabBarController *tabBar = (LDTTabBarController *)[[[[UIApplication sharedApplication] delegate] window] rootViewController];
+    [tabBar presentViewController:destNavVC animated:YES completion:nil];
+}
+
+                                                             
+#pragma mark - RCTBridgeModule
+
+RCT_EXPORT_MODULE();
+
+- (NSDictionary *)constantsToExport {
```

Comment:
This may be better suited for the `DSOUserManager`. I had a lot of trouble getting `self.bridge.addEvent` to work here to inform the `UserView.js` to reload the User's campaigns, which is why I switched to using `NSNotifications` and observers. [This SO thread](http://stackoverflow.com/questions/32879266/cant-send-an-event-from-main-thread-method-with-ios-and-react-native) goes into what may be the problem/solution, but NSNotifications seemed a little safer for our hybrid setup.